src/proxy-animation.js

```diff
@@ -912,6 +906,147 @@ function autoAlignStartTime(details) {
+  }
+  
+  if (timeline instanceof ScrollTimeline) {
+    return CSS.percent(100);
+  }
+
+  unsupportedTimeline(timeline);
+}
+
+function parseAnimationRange(timeline, value) {
+  const animationRange = {
+    start: getNormalStartRange(timeline),
+    end: getNormalEndRange(timeline),
+  };
+
+  if (!value)
+    return animationRange;
+
+  if (timeline instanceof ViewTimeline) {
+    // Format:
+    // <start-name> <start-offset> <end-name> <end-offset>
+    // <name> --> <name> 0% <name> 100%
+    // <name> <start-offset> <end-offset> --> <name> <start-offset>
+    //                                        <name> <end-offset>
+    // <start-offset> <end-offset> --> cover <start-offset> cover <end-offset>
+    // TODO: Support all formatting options once ratified in the spec.
+    const parts = value.split(' ');
+    const rangeNames = [];
+    const offsets = [];
+  
+    parts.forEach(part => {
+      if (part.endsWith('%'))
+        offsets.push(parseFloat(part));
+      else
+        rangeNames.push(part);
+    });
+  
+    if (rangeNames.length > 2 || offsets.length > 2 || offsets.length == 1) {
+      throw TypeError("Invalid time range or unsupported time range format.");
+    }
+  
+    if (rangeNames.length) {
+      animationRange.start.rangeName = rangeNames[0];
+      animationRange.end.rangeName = rangeNames.length > 1 ? rangeNames[1] : rangeNames[0];
+    }
+  
+    // TODO: allow calc() in the offsets
+    if (offsets.length > 1) {
+      animationRange.start.offset = CSS.percent(offsets[0]);
+      animationRange.end.offset = CSS.percent(offsets[1]);
+    }
+  
+    return animationRange;
+  }
+
+  if (timeline instanceof ScrollTimeline) {
+    // @TODO: Play nice with only 1 offset being set
+    // @TODO: Play nice with expressions such as `calc(50% + 10px) 100%`
+    const parts = value.split(' ');
+    if (parts.length != 2) {
+      throw TypeError("Invalid time range or unsupported time range format.");
+    }
+
+    animationRange.start = CSSNumericValue.parse(parts[0]);
+    animationRange.end = CSSNumericValue.parse(parts[1]);
+
+    return animationRange;
+  }
+
+  unsupportedTimeline(timeline);
+}
+
+function parseTimelineRangePart(timeline, value, position) {
+  if (!value || value === 'normal') return 'normal';
+
+  if (timeline instanceof ViewTimeline) {
+    // Extract parts from the passed in value.
+    let rangeName = 'cover'
+    let offset = position === 'start' ? CSS.percent(0) : CSS.percent(100)
+
+    // Author passed in something like `{ rangeName: 'cover', offset: CSS.percent(100) }`
+    if (value instanceof Object) {
+      if (value.rangeName !== undefined) {
+        rangeName = value.rangeName;
+      }
+
+      if (value.offset !== undefined) {
+        offset = value.offset;
+      }
+    }
+    // Author passed in something like `"cover 100%"`
+    else {
+      const parts = value.split(rangeNameRegExp).map(part => part.trim()).filter(Boolean);
```

Comment:
Sorry for bringing up https://github.com/flackr/scroll-timeline/pull/209 a lot 😅 

If it's merged, we could clean this up:
```js
const parts = splitIntoComponentValues(value);
```